Move fakefish from telco-operations repo

mvazquezc opened the pull request:
This PR migrates the FakeFish code from https://github.com/RHsyseng/telco-operations/tree/main/fakefish to this new repo.

Signed-off-by: Mario Vazquez <mavazque@redhat.com>

dtantsur commented on README.md:
```diff
@@ -0,0 +1,158 @@
+
+Users can take a look at the [dell_scripts](./dell_scripts/) folder where they can find the scripts we used for our testing with DELL hardware. Remember that these scripts are unsupported/unmaintained and should be taken as a reference, nothing else.
+
+A [Containerfile](./app/Containerfile) is included, so users can build their own container image out of it. We published an image using the bits of the first release at `quay.io/mavazque/fakefish:v0`.
```
We'll need to change this. I don't know if we should keep publishing images at all (rather than allowing the users to publish theirs). @rhjanders?

rhjanders replied:
@dtantsur here's my current thinking:
* initially, let's not worry about the images and assume users will make their own (this is to get to a good state asap)
* in general, perhaps it's beneficial to have images available to make getting started it as easy as possible
* however for that to work we need to think through how the images are built and kept up to date (better have no images available than to have old images) and that's adding complexity which is likely not needed at this stage.
WDYT?

dtantsur commented on README.md:
```diff
@@ -0,0 +1,158 @@
+       Use a production WSGI server instead.
+     * Debug mode: off
+     * Running on all addresses.
+       WARNING: This is a development server. Do not use it in a production deployment.
```
This is also something we need to address before we start offering this as a customization option. Probably by using gunicorn/uwsgi.

dtantsur commented on dell_scripts/mountcd.sh:
```diff
@@ -0,0 +1,18 @@
+ISO=${1}
+
+# Disconnect image just in case
+/opt/dell/srvadmin/bin/idracadm7 -r 192.168.1.10 -u root -p calvin remoteimage -d
```
We should either drop Dell scripts or make them more production-looking. At least provide the IP address and credentials.

racedo replied:
Note that this is meant only to illustrate how it would be done. Dell iDracs don't need this as they support Redfish directly as shipped with Metal3. Perhaps a good idea is to mention this in a comment in the files to make it very obvious for anyone reading them and not be tempted to put them to production by misunderstanding their purpose?

mvazquezc replied:
I force pushed changes to put a disclaimer in the README and in the scripts.

rhjanders replied:
Random idea: would there be value in talking to something easier to set up in dev than an actual BMC - say, running virsh commands or ipmitool? Or are we better off using examples that are closer to reality like the idracadm one?

racedo replied:
IMO I wouldn't move the iDrac examples to use virsh or ipmitool instead. These scripts with iDrac are relatable to potential users and illustrative of the how this can be done. If we want to reinforce it by adding another set of scripts showing virsh or ipmitool that's a nice to have. I don't find it necessary though, since these ones serve the purpose well already.

dtantsur reviewed (approved):
LGTM, letting @rhjanders take a look.

Some thoughts for future work inline.

rhjanders commented:
@mvazquezc Thank you for putting this together, excellent work.

My only suggestion (this is a bit of nitpicking really) at this stage would be reshuffling the files a little. Having looked through the logic I do think having custom_scripts and dell_scripts shipped separately is a good idea. Perhaps Containerfile should live in custom_scripts and ContainerfileDell should only live in dell_scripts, with no Containerfile* in app, WDYT? This may help prevent duplication of effort/code edits. Otherwise LGTM - I look forward to taking the code for a test!

mvazquezc commented:
hey @rhjanders I have done several modifications to address your comments.

- Removed custom_scripts from app folder.
- Created empty custom_scripts folder at the same level than dell_scripts
- Updated README accordingly
- Added Makefile to allow users build their own images (added instructions to the README)

Let me know if further changes are required, thanks!

rhjanders commented:
Thank you @mvazquezc this is great.
/lgtm
(not yet sure how this repo works but trying usual commands anyway :)
